### ISSUE TITLE
New Tutorial Feedback

### ISSUE THREAD

**Issue by jonfroehlich** [ProjectSidewalk/SidewalkWebpage#711]
In general, I think the (slightly) redesigned tutorial is better. I have two key concerns: (i) tutorial length and (ii) the introduction to walking is a bit confusing because of the size of the step (large distance) and because we don't animate the walk (like is typical); both contribute to a feeling of disorientation (i.e., "where am I?").

Some detailed feedback below.

My update text ideas are just suggestions and likely still need a bit of iteration; however, I do think they are improvements :) It would be great to hear feedback from the whole team on this but especially the grads.

1. Update text to: "...First, grab and drag the Street View image to look to the left" (i.e., don't we want to tell them the direction to look?)
![image](https://user-images.githubusercontent.com/1621749/27337713-3b84c0cc-55a1-11e7-9c42-9fef6f1fd469.png)

2. Update text to: "To label the curb ramp, click beneath the flashing yellow arrow."
![image](https://user-images.githubusercontent.com/1621749/27337781-6e99573e-55a1-11e7-9fd9-561fa7bf5509.png)

3. Update text to:  "Great Job! We are almost done. Now, let's learn how to walk and find issues. To look left, grab and drag the Street View image."
![image](https://user-images.githubusercontent.com/1621749/27337856-a6a6c710-55a1-11e7-9fd1-35f475680aac.png)

4. Taking a step in the tutorial is really abrupt. It's too bad you can't figure out how to perform the scene interpolation to make this smoother. It's actually a bit confusing because I don't know exactly where I am (the step was rather large too, so things look pretty different from the new location further adding to confusion).

5. Update text to: "Great! We just moved one step down the street. Now, we can look for more issues at this location. In this case, notice how there is a crosswalk without curb ramps. Ordinarily, you would label both sides of the street with a Missing Curb Ramp; however, let's finish learning about the rest of the interface instead." I think we could also split this up into two separate messages (where user has to click OK to go between them). 

First message: "Great! We just moved one step down the street. Now, we can look for more issues at this location. In this case, notice how there is a crosswalk without curb ramps." Button is labeled: "Yep, I see the missing curb ramps" (or something)

Second message: "Ordinarily, you would label the areas under the flashing arrows with a Missing Curb Ramp; however, we want to get you started on actual missions, so let's just finish this tutorial instead!"

6. This is brilliant. Big improvement. And I love the appear animation:

![image](https://user-images.githubusercontent.com/1621749/27338343-e38d18ea-55a2-11e7-93d4-4c96048c540b.png)

7. Update text to: "You can also follow the red line on this map. Your labels appear here too!"

![image](https://user-images.githubusercontent.com/1621749/27338374-f7140a86-55a2-11e7-93f4-369038d04ada.png)

8. I question whether we even need to onboard this part of the interface? Thoughts? If we do keep it, update text to: "You can track your mission progress and statistics here."
![image](https://user-images.githubusercontent.com/1621749/27338440-303d00ec-55a3-11e7-9017-fbf23a84ef34.png)

9. Consider replacing 'while walking' to 'while exploring.' Thoughts?

![image](https://user-images.githubusercontent.com/1621749/27338478-4b9a7e6e-55a3-11e7-83ab-c9afaa1cb745.png)

10. The first screen and last screen (non-GSV pane screens) need to be redesigned. They are hard to follow, read, and understand. I think I made this an issue somewhere else.

![image](https://user-images.githubusercontent.com/1621749/27338797-1c3c9980-55a4-11e7-97b6-301daba45882.png)

11. Consider changing the *first* mission text that someone ever receives with: "Your first mission is to..." -- this feels more exciting.

![image](https://user-images.githubusercontent.com/1621749/27338836-3380ed3a-55a4-11e7-87f6-309d8ab5183b.png)

12. I did not receive any further training upon my first mission (e.g., no pop-up about labels disappearing, no pop-up about walking, etc.).

13. Did we decide not to have the 'pan this way' arrow come up when the user pans in wrong direction. I did notice that we block the pan direction a bit--but do we think this is sufficient?

It would be great to get feedback from the whole team on the tutorial (from grads to ugrads) as well as to user test it.

**Comment by misaugstad** [comment]
>don't we want to tell them the direction to look?

I think that we should try to minimize the number of words, and once they have already successfully panned to the left twice, they should be able to use the visual cue of the hand grabbing the screen to know which direction to look in.

>Update text to: "Great Job! We are almost done. Now, let's learn how to walk and find issues. To look left, grab and drag the Street View image."

How about: "Great Job! We are almost done. Now, let's learn how to explore and find issues. Grab and drag the Street View Image."

>Consider replacing 'while walking' to 'while exploring.' Thoughts?

I definitely think that we should replace "walking" with "exploring" as much as possible. If the whole goal is to build a database of accessibility data so that people with mobility impairments can more easily navigate city streets, it seems silly to be using the word "walking", when many people with mobility impairments don't do that.

Though that is coming from the perspective of someone without a mobility impairment; maybe most individuals with mobility impairments use the word "walking" all the time and would never notice.

So those were comments on the use of "walking" over our tool in general. In this specific instance, I think that "explore" sounds more fun and exciting anyway, so I like the use of explore here regardless.

**Comment by jonfroehlich** [comment]
I completely agree that we should minimize text as much as possible. I was trying to balance concision with clarity. However, I agree that some of my suggestions are too wordy and that we should work together to cut them down (as well as other messages in the interface).

**Comment by manaswis** [comment]
**Pts 2, 7, 8, 9**: Made the minor adjustments to the text for other instructions. 

@jonfroehlich For the placement of "Your labels appear here too!":
![image](https://user-images.githubusercontent.com/2873216/27386150-eec4bc66-5662-11e7-8043-b2066d16b5f1.png)
Ignore the spacing issue in the above screenshot.

OR

![image](https://user-images.githubusercontent.com/2873216/27386946-54ce2310-5665-11e7-8012-9957d42ffce8.png)



**Pts 1 and 3**:
> Update text to: "...First, grab and drag the Street View image to look to the left" (i.e., don't we want to tell them the direction to look?)

As @misaugstad mentioned, from the first two corners, the users should now know to pan left. So kept these messages the same without the pan direction. 

**Pt 5**: Updated After Walk Message (tweaked a bit here and there):
 
1. @jonfroehlich For this, what do you think about "Yes! I see the missing curb ramps." instead of the current "Yep, I see the missing curb ramps!". Its a slight modification.
![image](https://user-images.githubusercontent.com/2873216/27385472-cff9cf6c-5660-11e7-97f9-13d44cb5337b.png)

2. @jonfroehlich For this, what do you think about "real missions" instead of "actual missions"?
![image](https://user-images.githubusercontent.com/2873216/27385525-0021345a-5661-11e7-893d-c45556b8e71e.png)

**Pt 8**: 
> I question whether we even need to onboard this part of the interface? Thoughts? 

I think we should have it. It feels more complete with it.

**Pt 11**:
> Consider changing the first mission text that someone ever receives with: "Your first mission is to..." -- this feels more exciting.

I agree. I will make another issue for it.

**Pt 12**:
> I did not receive any further training upon my first mission (e.g., no pop-up about labels disappearing, no pop-up about walking, etc.).

There is a bug. Those messages are not showing up after onboarding. Needs to be looked into. Creating separate issue for it.

**Pt 13**:
> Did we decide not to have the 'pan this way' arrow come up when the user pans in wrong direction. I did notice that we block the pan direction a bit--but do we think this is sufficient?

I wanted to see what you thought without it. I have reduced the panning significantly in every step. It has been reduced such that the user can see the current stage corner and the previous stage. For the after walk step, the user can pan a bit more on both sides of the road beyond the arrows.

**Comment by jonfroehlich** [comment]
It would be most efficient for me to talk to you about all of these things in person. :)

I have an idea for 
![image](https://user-images.githubusercontent.com/1621749/27388170-bf01f65a-5668-11e7-9490-600da813c253.png)
that is far simpler, I think.

**Comment by maddalihanumateja** [comment]
1. I do agree that tutorial might be slightly lengthy however most of the time in the tutorial is spent on getting a 360 degree view and labeling curb ramps around the intersection. I think I remember previous discussions mentioning it was important for the users to commit this behavior to memory near intersections. I think over the shoulder testing will give us a better idea about the perceived tutorial length.

2. The text messages feel relatively better than those of the older version. 
        - I think that, in general, when the user has to label something below a flashing arrow the text message should be more on the lines of "Click on the X below the flashing arrow." 
        - When we are labeling missing curb ramps with severity 3 then assigning a severity of 4 or 5 gives the same message as the one for 1 or 2 : "This is a slightly severe problem". 
<img width="1262" alt="screen shot 2017-06-21 at 9 34 51 am" src="https://user-images.githubusercontent.com/3580069/27388326-29619154-5669-11e7-90c2-1a469c771904.png">
I think we could have a more agnostic message like in the other prompts. "This looks more like a 3."
        - The "grab and drag" message still seemed confusing to my folks. I was able to get the desired behavior by saying "look to the left either by using the arrow key on your keyboard or by by clicking your mouse on the screen and dragging it.". What do you guys think?

3. Label placement tolerance
<img width="1239" alt="screen shot 2017-06-21 at 9 38 08 am" src="https://user-images.githubusercontent.com/3580069/27388606-fabba3de-5669-11e7-8afa-af5813989969.png">
 may need to be adjusted but I dont think it is that high of a priority

4. I might have mentioned this to manaswi before. This image is a good example of missing curb ramps but it slightly siturbs the flow of the tutorial when the text message is saying "You would label them ... However lets move on". (The labeling icons are disabled so I cant label them anyway). I think this message can be skipped but I'm going to go by group opinion.
<img width="1271" alt="screen shot 2017-06-21 at 9 34 37 am" src="https://user-images.githubusercontent.com/3580069/27388551-d353c2ae-5669-11e7-8d1c-a54b052ff835.png">

I was testing with Firefox and Chrome on my Mac and I didn't run into any browser specific issues for the onboarding.

**Comment by manaswis** [comment]
> 2. The text messages feel relatively better than those of the older version. 
>        
>  - I think that, in general, when the user has to label something below a flashing arrow the text message should be more on the lines of "Click on the X below the flashing arrow." Rather than just "Click beneath the flashing arrow".
>         
> - When we are labeling missing curb ramps with severity 3 then assigning a severity of 4 or 5 gives the same message as the one for 1 or 2 : "This is a slightly severe problem". I think we could have a more agnostic message like in the other prompts. "This looks more like a 3."
>
>         
> - The "grab and drag" message still seemed confusing to my folks. I was able to get the desired behavior by saying "look to the left either by using the arrow key on your keyboard or by by clicking your mouse on the screen and dragging it.". What do you guys think?
> 

I like these suggestions. I will work on updating it. Grab and drag needs a bit more thinking. We don't want too many words.

**Comment by manaswis** [comment]
Closed via PR #750 and #748

**Comment by jonfroehlich** [comment]
> Jon: Did we decide not to have the 'pan this way' arrow come up when the user pans in wrong direction. I did notice that we block the pan direction a bit--but do we think this is sufficient?
> 
> Manaswi: I wanted to see what you thought without it. I have reduced the panning significantly in every step. It has been reduced such that the user can see the current stage corner and the previous stage. For the after walk step, the user can pan a bit more on both sides of the road beyond the arrows.

It looks like this continues to be a problem judging by the comment below. (However, perhaps this is a different issue. The user left their email addie, so we could follow-up). @manaswis.

![image](https://user-images.githubusercontent.com/1621749/33336764-c47dc966-d425-11e7-8485-b18c85270552.png)

**Comment by manaswis** [comment]
Okay. I will follow up with the person to know what is the problem exactly.

**Comment by jonfroehlich** [comment]
Please cc me
